x-pack/plugins/apm/server/lib/service_map/get_service_map.ts

```diff
@@ -76,21 +75,17 @@ async function getServicesData(options: IEnvOptions) {
     setup: { ...setup, uiFiltersES: [] },
   });
 
-  const { filter } = projection.body.query.bool;
+  const serviceNameFilter = options.serviceName
+    ? [{ term: { [SERVICE_NAME]: options.serviceName } }]
+    : [];
 
-  const params = mergeProjection(projection, {
+  const params = {
     body: {
+      index: projection.index,
       size: 0,
       query: {
         bool: {
-          ...projection.body.query.bool,
-          filter: options.serviceName
-            ? filter.concat({
-                term: {
-                  [SERVICE_NAME]: options.serviceName,
-                },
-              })
-            : filter,
+          filter: [...projection.body.query.bool.filter, ...serviceNameFilter],
```

Comment:
This is a good example of the confusion I get from the projection:
 - we have a higher order function `mergeProjection` that merges some stuff
 - then we have `projection.body.query.bool` which is spreaded onto `bool`
 - then we have a specific part of the projection being merged with the full `bool` projection + some other stuff

What takes precendence? Is `...projection.body.query.bool` really needed?

Reply:
`mergeProjection` = `mergePlainObject`, e.g. it doesn't merge arrays. If you merge arrays, you can't remove items. Arguably the same applies for merging objects, but that feels more natural. If I speak for myself, in some cases I use explicit spreading so that the shape of the request (ie the result of `mergeProjection`) is more clear. (Not a argument for or against projections, just an explanation).